shipyard_blueprints/dropbox/shipyard_dropbox/cli/upload.py

```diff
@@ -171,85 +78,87 @@ def download_dropbox_file(file_name, client, destination_file_name=None):
+        if args.source_file_name_match_type == "exact_match":
+            destination_name = shipyard.determine_destination_full_path(
                 destination_folder_name=destination_folder_name,
                 destination_file_name=args.destination_file_name,
-                source_full_path=file_name,
-                file_number=index + 1,
+                source_full_path=source_full_path,
             )
 
-            if not file_name.startswith("/"):
-                file_name = f"/{file_name}"
-            print(f"Downloading file {index+1} of {len(matching_file_names)}")
             download_dropbox_file(
-                file_name=file_name,
+                file_name=(
+                    source_full_path
+                    if source_full_path.startswith("/")
+                    else f"/{source_full_path}"
+                ),
                 client=client,
                 destination_file_name=destination_name,
             )
-    else:
-        destination_name = determine_destination_name(
-            destination_folder_name=destination_folder_name,
-            destination_file_name=args.destination_file_name,
-            source_full_path=source_full_path,
-        )
+        if args.source_file_name_match_type == "regex_match":
+            file_names = find_dropbox_file_names(
+                client=client, prefix=source_folder_name
+            )
+            matching_file_names = shipyard.find_all_file_matches(
+                file_names, re.compile(source_file_name)
+            )
+            if number_of_matching_files := len(matching_file_names) == 0:
+                logger.error(f"No files found with the regex {source_file_name}")
+                sys.exit(1)
 
-        if not source_full_path.startswith("/"):
-            source_full_path = f"/{source_full_path}"
-        download_dropbox_file(
-            file_name=source_full_path,
-            client=client,
-            destination_file_name=destination_name,
-        )
+            logger.info(
+                f"{len(matching_file_names)} files found. Preparing to download..."
+            )
+
+            for index, file_name in enumerate(matching_file_names, start=1):
+                destination_name = shipyard.determine_destination_full_path(
+                    destination_folder_name=destination_folder_name,
+                    destination_file_name=args.destination_file_name,
+                    source_full_path=file_name,
+                    file_number=index,
+                )
+
+                logger.info(f"Uploading file {index} of {number_of_matching_files}")
+                download_dropbox_file(
+                    file_name=(
+                        file_name if file_name.startswith("/") else f"/{file_name}"
+                    ),
+                    client=client,
+                    destination_file_name=destination_name,
+                )
+    except Exception as e:
```

Comment:
Same comment as before, where upon completion have one more additional log statement that lets the user know that their file has been successfully uploaded to Dropbox